### PR TITLE
Master

### PR DESCRIPTION
**Pull request by fscarmen**

### CODE REVIEW

File: release.sh [hunk 1]
```diff
@@ -32,24 +32,26 @@ fi
...
+  echo "Event: ${GITHUB_EVENT_NAME}"
 elif [ ${GITHUB_EVENT_NAME} == 'push' ]; then
-    echo "Event: ${GITHUB_EVENT_NAME}"
+  echo "Event: ${GITHUB_EVENT_NAME}"
 elif [ ${GITHUB_EVENT_NAME} == 'workflow_dispatch' ]; then
-    echo "Event: ${GITHUB_EVENT_NAME}"
+  echo "Event: ${GITHUB_EVENT_NAME}"
 elif [ ${GITHUB_EVENT_NAME} == 'workflow_run' ]; then
-    echo "Event: ${GITHUB_EVENT_NAME}"
+  echo "Event: ${GITHUB_EVENT_NAME}"
```
Comment (wangyoucao577) [inline release.sh:41]: Are you using 2-space indent rather than 4?

File: release.sh [hunk 2]
```diff
@@ -32,24 +32,26 @@ fi
...
+elif [ ${GITHUB_EVENT_NAME} == 'schedule' ] && [ ${INPUT_ALLOW_SCHEDULE^^} == 'TRUE' ]; then
```
Comment (wangyoucao577) [inline release.sh:42]: In my option, add the `allow_schedule` arg is a little bit redundant and complicated, just add one more line here for `schedule` check it should be good enough.

**Review by wangyoucao577 — COMMENTED** [review]
One more

**Comment by fscarmen** [comment]
I don't use it enough to understand your reasons for designing disallowed schedules, and I'm guessing it must be that other users have had problems with it before and that's why you're making this restriction. So I added allow_schedule to preserve the original restriction to the greatest extent possible, but of course it's a better option to just lift the restriction if it's something you've assessed as safe.

```
# prompt error if non-supported event
if egrep -q 'release|push|workflow_dispatch|workflow_run|schedule'; then
  echo "Event: ${GITHUB_EVENT_NAME}"
else
  echo "Unsupport event: ${GITHUB_EVENT_NAME}!  Allow: release | push | workflow_dispatch | workflow_run | schedule"
  exit 1
fi
```

**Comment by wangyoucao577** [comment]
> I don't use it enough to understand your reasons for designing disallowed schedules, and I'm guessing it must be that other users have had problems with it before and that's why you're making this restriction. So I added allow_schedule to preserve the original restriction to the greatest extent possible, but of course it's a better option to just lift the restriction if it's something you've assessed as safe.
> 
> ```
> # prompt error if non-supported event
> if egrep -q 'release|push|workflow_dispatch|workflow_run|schedule'; then
>   echo "Event: ${GITHUB_EVENT_NAME}"
> else
>   echo "Unsupport event: ${GITHUB_EVENT_NAME}!  Allow: release | push | workflow_dispatch | workflow_run | schedule"
>   exit 1
> fi
> ```

This looks better and good enough. At very early days, there're people uses the action by undefined event, so I added this restriction to make the error message clear, so that people will know how to fix it.

**Pull request closed without merging**